File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/Sticker.java
package ja.burhanrashid52.photoeditor;

import android.graphics.Bitmap;
import android.view.View;
import android.widget.ImageView;
import android.widget.RelativeLayout;

/**
 * Created by Burhanuddin Rashid on 14/05/21.
 *
 * @author <https://github.com/burhanrashid52>
 */
class Sticker extends Graphic {

    private final MultiTouchListener mMultiTouchListener;
    private final OnPhotoEditorListener mOnPhotoEditorListener;
    private final RelativeLayout mCanvasView;
    private final PhotoEditorViewState mViewState;
    private ImageView imageView;

    public Sticker(RelativeLayout canvasView,
                   PhotoEditorView photoEditorView,
                   MultiTouchListener multiTouchListener,
                   PhotoEditorViewState viewState,
                   OnPhotoEditorListener onPhotoEditorListener,
                   GraphicManager graphicManager
    ) {
        super(photoEditorView.getContext(), graphicManager);
        mCanvasView = canvasView;
        mViewState = viewState;
        mOnPhotoEditorListener = onPhotoEditorListener;
        mMultiTouchListener = multiTouchListener;
        setupGesture();
    }

    void buildView(Bitmap desiredImage) {
        imageView.setImageBitmap(desiredImage);
    }

    private void setupGesture() {
        MultiTouchListener.OnGestureControl onGestureControl = buildGestureController(
                mCanvasView,
                mViewState,
                mOnPhotoEditorListener
        );
        mMultiTouchListener.setOnGestureControl(onGestureControl);
        View rootView = getRootView();
        rootView.setOnTouchListener(mMultiTouchListener);
    }


    @Override
    ViewType getViewType() {
        return ViewType.IMAGE;
    }

    @Override
    int getLayoutId() {
        return R.layout.view_photo_editor_image;
    }

    @Override
    void setupView(View rootView) {
        imageView = rootView.findViewById(R.id.imgPhotoEditorImage);
    }
}

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/Sticker.kt
package ja.burhanrashid52.photoeditor

import android.widget.RelativeLayout
import android.graphics.Bitmap
import android.view.View
import android.widget.ImageView

/**
 * Created by Burhanuddin Rashid on 14/05/21.
 *
 * @author <https:></https:>//github.com/burhanrashid52>
 */
internal class Sticker(
    private val mCanvasView: RelativeLayout,
    private val mPhotoEditorView: PhotoEditorView,
    private val mMultiTouchListener: MultiTouchListener,
    private val mViewState: PhotoEditorViewState,
    private val mOnPhotoEditorListener: OnPhotoEditorListener,
    graphicManager: GraphicManager?
) : Graphic(
    context = mPhotoEditorView.context,
    graphicManager = graphicManager,
    viewType = ViewType.IMAGE,
    layoutId = R.layout.view_photo_editor_image
) {
    private var imageView: ImageView? = null
    fun buildView(desiredImage: Bitmap?) {
        imageView?.setImageBitmap(desiredImage)
    }

    private fun setupGesture() {
        val onGestureControl = buildGestureController(
            mCanvasView,
            mViewState,
            mOnPhotoEditorListener
        )
        mMultiTouchListener.setOnGestureControl(onGestureControl)
        val rootView = rootView
        rootView.setOnTouchListener(mMultiTouchListener)
    }

    override fun setupView(rootView: View) {
        imageView = rootView.findViewById(R.id.imgPhotoEditorImage)
    }

    init {
        setupGesture()
    }
}